source/views/sis/course-search/row.js

```diff
@@ -22,20 +23,26 @@ export class CourseRow extends React.PureComponent<Props> {
 
 	render() {
 		const {course} = this.props
-		const times = course.times
-			? course.times.map(time => {
-					// Splits days and times  ex. 'MWF 0800-0900' -> ['MWF', '0800-0900PM']
-					let array = time.split(/\s/)
-					let cleanedTime = findTime(array[1])
+		const groupings = convertTimeStringsToOfferings(
+			{times: course.times},
+			{groupBy: 'sis'},
+		)
+		const formattedGroupings = groupings.map(grouping => {
+			const times = grouping.times
+				.map(time => {
 					const start = moment
-						.tz(cleanedTime.start, 'hmm', CENTRAL_TZ)
-						.format('h:mm A')
-					const end = moment
-						.tz(cleanedTime.end, 'hmm', CENTRAL_TZ)
+						.tz(time.start, 'hmm', CENTRAL_TZ)
 						.format('h:mm A')
-					return `${array[0]} ${start} - ${end}`
+					const end = moment.tz(time.end, 'hmm', CENTRAL_TZ).format('h:mm A')
+					return `${start} - ${end}`
 				})
-			: null
+				.join(', ')
+			let days = grouping.days.map(day => formatDayAbbrev(day)).join('')
+			if (days === 'MTWThF') {
```

Comment:
I'll warn you that SIS doesn't always put them in order – https://github.com/StoDevX/sis-timestring-parser/blob/master/src/__tests__/find-days.test.js#L35 is from an actual course entry.

Reply:
I suppose that you could sort the days before turning it into a string - 

```js
const dayOrder = ['Mo', 'Tu', 'We', 'Th', 'Fr']
let days = _.sortBy(grouping.days, d => dayOrder.indexOf(d)).map(formatDayAbbrev).join('')
```

Reply:
☹️